app/controllers/AddClosureMetadataController.scala

```diff
@@ -2,50 +2,92 @@ package controllers
 
 import auth.TokenSecurity
 import configuration.{GraphQLConfiguration, KeycloakConfiguration}
-import graphql.codegen.GetCustomMetadata.customMetadata.CustomMetadata
-import controllers.util.CustomMetadataUtils
 import controllers.util.CustomMetadataUtils.FieldValues
+import graphql.codegen.GetCustomMetadata.customMetadata.CustomMetadata
+import controllers.util.{CustomMetadataUtils, DynamicFormUtils}
 import org.pac4j.play.scala.SecurityComponents
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, Request}
 import services.{ConsignmentService, CustomMetadataService}
-import viewsapi.Caching.preventCaching
+import play.api.cache._
 
 import java.util.UUID
 import javax.inject.Inject
 import scala.collection.immutable.ListSet
-import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
 
 class AddClosureMetadataController @Inject()(val controllerComponents: SecurityComponents,
                                              val graphqlConfiguration: GraphQLConfiguration,
                                              val keycloakConfiguration: KeycloakConfiguration,
                                              val consignmentService: ConsignmentService,
-                                             val customMetadataService: CustomMetadataService)
+                                             val customMetadataService: CustomMetadataService,
+                                             val cache: AsyncCacheApi)
                                             (implicit val ec: ExecutionContext) extends TokenSecurity with I18nSupport {
 
   def addClosureMetadata(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) {
     implicit request: Request[AnyContent] =>
       for {
         consignmentRef <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
-        customMetadata <- customMetadataService.getCustomMetadata(consignmentId, request.token.bearerAccessToken)
-        customMetadataUtils = CustomMetadataUtils(customMetadata)
-        propertyName = Set("ClosureType")
-        value = "closed_for"
+        orderedFieldsForForm <- { // currently not ordered, but will be soon; please delete this comment when it is
+          cache.set(s"$consignmentId", consignmentRef, 1.hour)
+          getDefaultFieldsForForm(consignmentId, request)
+        }
+      }
+        yield Ok(views.html.standard.addClosureMetadata(consignmentId, consignmentRef, orderedFieldsForForm, request.token.name))
+  }
+
+  def addClosureMetadataSubmit(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) {
```

Comment:
I guess you'll add tests in for this endpoint once everything is completed?

Reply:
Yes.